components/pages/submission/Environmental/NewSubmissions/index.tsx

```diff
@@ -46,44 +47,55 @@ const NewSubmissions = (): ReactElement => {
 	const [thereAreFiles, setThereAreFiles] = useState(false);
 	const [uploadError, setUploadError] = useState(noUploadError);
 	const [validationState, validationDispatch] = useReducer(validationReducer, validationParameters);
-	const { oneTSV, oneOrMoreFasta, readyToUpload } = validationState;
+	const { oneOrMoreCsv, readyToUpload } = validationState;
 
-	const { awaitingResponse, fetchMuseData } = useMuseData('NewSubmissions');
+	const { awaitingResponse, submitData } = useEnvironmentalData('NewSubmissions');
 
 	const handleSubmit = () => {
 		if (thereAreFiles && token && userHasWriteScopes) {
 			const formData = new FormData();
 
-			// if many TSV are available, submit only the first one along with all fastas
-			const selectedTSV = oneTSV.slice(-1)[0];
-			formData.append('files', selectedTSV, selectedTSV.name);
-			oneOrMoreFasta.forEach((fasta) => formData.append('files', fasta, fasta.name));
+			oneOrMoreCsv.forEach((csvFile) => {
+				// TODO: Fine grained permissions. Check if user has permission to upload Environmental data to this organization
+				// Taking the organization name from the filename
+				const organizationName = csvFile.name.split('.')[0].toUpperCase();
```

Comment:
not asking for changes, as the TODO note mentions this is work in progress; but still going to point out how the chain seems brittle here.